mlflow/server/js/src/reducers/Reducers.js

```diff
@@ -156,19 +149,28 @@ export const getRunTags = (runUuid, state) => {
 };
 
 const tagsByRunUuid = (state = {}, action) => {
+  const tagArrToObject = (tags) => {
+    const tagObj = {};
+    tags.forEach((tag) => tagObj[tag.key] = RunTag.fromJs(tag));
+    return tagObj;
+  };
   switch (action.type) {
     case fulfilled(GET_RUN_API): {
       const runInfo = RunInfo.fromJs(action.payload.run.info);
-      return amendTagsByRunUuid(state, action.payload.run.data.tags, runInfo.getRunUuid());
+      const tags = action.payload.run.data.tags | [];
+      const runUuid = runInfo.getRunUuid();
+      const newState = {...state};
+      newState[runUuid] = tagArrToObject(tags);
+      return newState;
     }
     case fulfilled(SEARCH_RUNS_API): {
       const runs = action.payload.runs;
       let newState = { ...state };
       if (runs) {
         runs.forEach((rJson) => {
-          const run = Run.fromJs(rJson);
-          newState = amendTagsByRunUuid(
-              newState, rJson.data.tags, run.getInfo().getRunUuid());
+          const runUuid = rJson.info.run_uuid;
```

Comment:
Creating the immutable.js object via `fromJs` is surprisingly slow - removing that seems to speed this code block up 2x